app/logic/events.py

```diff
@@ -501,3 +507,18 @@ def getEventRsvpCount(eventId):
         Returns the number of RSVP'd participants for a given eventId.
     """
     return len(EventRsvp.select().where(EventRsvp.event_id == eventId))
+
+def copyRsvp(priorEvent, newEvent):
+    rsvpInfo = list(EventRsvp.select().where(EventRsvp.event == priorEvent).execute())
+    rsvpCopies = 0
+    for student in rsvpInfo:
+        newRsvp = EventRsvp(
+                user = student.user,
+                event = newEvent,
+                rsvpWaitlist = student.rsvpWaitlist
+            )
+        newRsvp.save()
+        rsvpCopies=len(rsvpInfo)
+
+    if rsvpCopies:
+        createRsvpLog(newEvent, f"Copied {rsvpCopies} Rsvp's from {priorEvent.name} to {newEvent.name}")
```

Comment:
The name of the renewed event should link to the new event like how we link to newly created/canceled events. 

![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/26262045/cea8baad-f4c7-4cdc-9e2b-a7b944b7e49f)

Reply:
I have resolved this.